yaml.qmd

```diff
@@ -0,0 +1,874 @@
+toc: true           # A boolean, `true`/`false`
+toc-depth: 2        # A number
+---
+```
+
+The only values recognized as boolean are `true` and `false`, numbers can be single integers (`2`), decimal (`2.1`) or scientific (`2.1e+2`), and anything else is treated as a string.
+
+### Quotes
+
+Strings don't require quotes, but you may need them if your string contains characters that have special meaning in YAML (e.g. `#`, `:`, etc.), or is something that would otherwise be interpreted as a number or boolean:
+
+``` {.yaml filename="document.qmd"}
+---
+title: "Quotes: and things that need them"  
+author: "#quarto"
+subtitle: "100"
+---
+```
+
+In YAML double quoted strings (`"`) allow escape characters in them, whereas single quoted strings (`'`) are interpreted literally:
+
+``` {.yaml filename="document.qmd"}
+---
+title: "Quotes: \U0001F4E8"
+subtitle: 'Quotes: \U0001F4E8'
+---
+```
+
+Q: where does the `\U` go in `subtitle` go?
+YAML might be interpreting literally but Quarto/Pandoc might escape somewhere?
+
+### Long strings
+
+For strings that are longer than you'd comfortably put on a single line, or need explicit line breaks, you can use what YAML calls block scalars.
+
+There are two styles: literal and folded.
+
+Literal block scalars are indicated by a pipe (`|`), and line breaks are considered significant, whereas folded block scalars are indicated by a greater than symbol (`>`), and line breaks are collapsed to spaces.
+
+In both cases, start your string on a new line indented under the key:
+
+::: {layout-ncol="2"}
+``` {.yaml filename="literal.qmd"}
+---
+title: |
+  A title
+
+  spanning two lines
+---
+```
+
+``` {.yaml filename="folded.qmd"}
+---
+title: >
+  A title
+
+  collapsed into one line
+---
+```
+:::
+
+::: {#fig-block-strings layout-ncol="2"}
+![Rendered title using literal (`|`) block scalar](images/yaml-block-literal.png){#fig-block-strings-literal .border fig-alt="A screenshot of a title that reads 'A title' then a line break followed by 'spanning two lines'."}
+
+![Rendered title using folded (`>`) block scalar](images/yaml-block-folded.png){#fig-block-strings-folded .border fig-alt="A screenshot of a title that reads 'A title collapsed into one line' all on a single line."}
+:::
+
+Depending on where the string value is used, line breaks may or may not mean anything to Quarto.
+For instance, when interpreted as markdown, a single line break doesn't usually result in anything in the output.
+However, a double line break will be interpreted as a new paragraph.
+
+### Using `default`
+
+You'll come across examples where the value of a Quarto option is `default`, like when you are specifying multiple formats:
+
+``` {.yaml filename="document.qmd"}
+---
+format:
+  html: default
+  pdf: default
+---
+```
+
+The value `default` doesn't have a special YAML meaning--- it's something Quarto adds for certain options where the value is usually another mapping, but you just want the default values.
```

Comment:
```suggestion
The value `default` doesn't have a special YAML meaning--- it's something Quarto adds for certain options where the value is usually another mapping, but you just want the default values for all subfields.
```

Reply:
I'm not sure if subfield is the right term here.

Reply:
Switched to "properties"